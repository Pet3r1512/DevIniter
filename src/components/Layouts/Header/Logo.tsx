import { cn } from "@/utils/utils";
import Image from "next/image";

export default function Logo({ className }: { className?: string }) {
  return (
    <Image
      src={"/images/logos/FullLogo.png"}
      alt="Logo"
      width={0}
      height={0}
      sizes="100vw"
      quality={85}
      priority
      data-testid="Logo"
      className={cn("lg:h-12 w-auto h-10 max-w-1/3 lg:max-w-none", className)}
    />
  );
}
