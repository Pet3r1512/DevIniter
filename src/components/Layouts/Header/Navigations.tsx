import Link from "next/link";

export default function Navigation() {
  const navs = [
    {
      name: "Documents",
      href: "/document",
    },
    {
      name: "Contribution",
      href: "/contribution",
    },
  ];
  return (
    <nav data-testid="navbar" className="hidden lg:flex items-center gap-x-5">
      {navs.map((nav, index) => {
        return (
          <Link
            key={index}
            href={nav.href}
            className="lg:hover:text-primary transition-all duration-150 ease-linear lg:text-lg"
          >
            {nav.name}
          </Link>
        );
      })}
    </nav>
  );
}
