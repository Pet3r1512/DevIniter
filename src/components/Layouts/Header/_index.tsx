import Github from "./Github";
import ThemeToggle from "../ThemeToggle";
import Sidebar from "./Sidebar/_index";
import Link from "next/link";
import { ArrowUpRight } from "lucide-react";
import dynamic from "next/dynamic";
import LogoSkeleton from "../Skeletons/LogoSkeleton";

const DynamicLogo = dynamic(() => import("./Logo"), {
  loading: () => <LogoSkeleton />,
  ssr: false,
});

const DynamicNavBar = dynamic(() => import("./Navbar"), {
  loading: () => <header className="lg:w-[500px] lg:h-9 relative" />,
  ssr: false,
});

export default function Header() {
  return (
    <header
      data-testid="header"
      className="flex items-center justify-between py-5 px-5 lg:px-0 sticky top-0 z-30 !backdrop-filter !backdrop-blur-xl"
    >
      <div className="max-w-[1440px] w-full mx-auto flex items-center justify-between">
        <div className="flex items-center gap-x-8">
          <DynamicLogo />
          <DynamicNavBar />
        </div>
        <div className="size-6 invisible" />
        <div className="lg:flex items-center gap-x-5 hidden">
          <Link
            href="https://github.com/Pet3r1512/DevIniter/issues/new?labels=enhancement&template=feature-request---.md"
            target="_blank"
            className="text-lg flex"
          >
            Contributing
            <ArrowUpRight size={14} />
          </Link>
          <Link href="/release" className="text-lg flex">
            Release Notes
            <ArrowUpRight size={14} />
          </Link>
          <VerticalLine />
          <ThemeToggle />
          <Github size={20} />
        </div>
        <Sidebar />
      </div>
    </header>
  );
}

function VerticalLine() {
  return (
    <div
      data-testid="vt"
      className="border-l-2 border-black-main dark:border-white h-[25px]"
    ></div>
  );
}
