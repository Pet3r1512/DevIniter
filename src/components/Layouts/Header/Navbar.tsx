"use client";

import * as React from "react";
import Link from "next/link";
import {
  NavigationMenu,
  NavigationMenuContent,
  NavigationMenuItem,
  NavigationMenuLink,
  NavigationMenuList,
  NavigationMenuTrigger,
} from "@/components/ui/navigation-menu";
import { cn } from "@/utils/utils";
import ComingSoon from "../Badges/ComingSoon";
import NewBadge from "../Badges/NewBadge";
import Logo from "./Logo";
import { useSmoothScroll } from "@/hooks/useSmoothScroll";
import { useInViewStore } from "@/stores/useInViewStore";
import Image from "next/image";

const badgeClassName =
  "relative ml-1.5 lg:ml-0 lg:top-[-12px] text-sm px-1.5 py-1";

export const BadgePattern: Record<string, React.ReactNode> = {
  new: <NewBadge className={badgeClassName} />,
  coming: <ComingSoon className={badgeClassName} />,
};

const templates: {
  title: string;
  image: string;
  href: string;
  description: string;
}[] = [
  {
    title: "Next.js",
    image: "/images/templates/nextjs-deviniter.png",
    href: "/docs/template_structure/nextjs",
    description:
      "A fullstack React framework for building scalable web applications.",
  },
  {
    title: "Vite",
    image: "/images/templates/Vite_logo.png",
    href: "/docs/template_structure/vite",
    description: "A fast and modern build tool for React applications.",
  },
  {
    title: "Vitestack",
    image: "/images/templates/ViteStack.png",
    href: "",
    description: "A modern and stunning frontend web application template.",
  },
];

const explores: {
  title: string;
  href: string;
  isHash?: boolean;
  description: string;
}[] = [
  {
    title: "Live Demo",
    href: "#demo",
    isHash: true,
    description:
      "Visit DevIniter templates in action with our interactive demo",
  },
  {
    title: "Tech Stacks",
    href: "#techstacks",
    isHash: true,
    description: "Explore the powerful technologies behind our templates",
  },
  {
    title: "Templates Features",
    href: "/features",
    description:
      "Discover all the built-in features and capabilities of our templates",
  },
];

const triggerClassName = "!bg-transparent dark:text-white lg:text-lg";
const itemClassName =
  "isolate ring-1 dark:bg-black/50 bg-white/75 ring-black/5 lg:hover:scale-105 lg:dark:hover:bg-black lg:hover:bg-white transition-all duration-200 ease-linear";

export default function Navbar() {
  const { handleHashScroll } = useSmoothScroll();
  const { isAnimationInView } = useInViewStore();

  return (
    <NavigationMenu
      data-testid="navbar"
      className="mt-1 hidden lg:block lg:w-[500px] lg:h-9"
    >
      <NavigationMenuList>
        <NavigationMenuItem>
          <NavigationMenuTrigger
            className={cn(
              triggerClassName,
              isAnimationInView ? "text-white" : "text-black"
            )}
          >
            Getting Started
          </NavigationMenuTrigger>
          <NavigationMenuContent>
            <ul className="grid gap-3 p-4 md:w-[400px] lg:w-[500px] lg:grid-cols-[.75fr_1fr] shadow-2xl dark:bg-black-main bg-white border-none !outline-none">
              <li className="row-span-3">
                <NavigationMenuLink asChild>
                  <Link
                    className="flex h-full w-full select-none flex-col justify-end rounded-md bg-gradient-to-b from-muted/50 to-muted p-6 no-underline outline-none focus:shadow-md"
                    href="/"
                  >
                    <Logo />
                    <div className="flex-1" />
                    <p className="text-sm leading-tight text-muted-foreground">
                      Blazing Starter Templates for Developers
                    </p>
                  </Link>
                </NavigationMenuLink>
              </li>
              <ListItem href="/docs/introduction" title="Introduction">
                {
                  "Learn about DevIniter's core concepts, features and the technology stack we use."
                }
              </ListItem>
              <ListItem href="/docs/get_started" title="Documents">
                Step-by-step guide to set up your project and understand the
                project structure.
              </ListItem>
              <ListItem href="/docs/installation" title="Installation">
                Choose your favorite template and download it now.
              </ListItem>
            </ul>
          </NavigationMenuContent>
        </NavigationMenuItem>
        <NavigationMenuItem>
          <NavigationMenuTrigger
            className={cn(
              triggerClassName,
              isAnimationInView ? "text-white" : "text-black"
            )}
          >
            Template Structures
          </NavigationMenuTrigger>
          <NavigationMenuContent>
            <ul className="grid w-[400px] gap-3 p-4 md:w-[500px] md:grid-cols-2 lg:w-[600px] dark:bg-black-main bg-white">
              {templates.map((template) => (
                <ListItem
                  key={template.title}
                  title={template.title}
                  image={template.image}
                  href={template.href}
                >
                  {template.description}
                </ListItem>
              ))}
            </ul>
          </NavigationMenuContent>
        </NavigationMenuItem>
        <NavigationMenuItem>
          <NavigationMenuTrigger
            className={cn(
              triggerClassName,
              isAnimationInView ? "text-white" : "text-black"
            )}
          >
            Explore More
          </NavigationMenuTrigger>
          <NavigationMenuContent>
            <ul className="grid w-[400px] gap-3 p-4 md:w-[500px] md:grid-cols-2 lg:w-[600px] dark:bg-black-main bg-white">
              {explores.map((explore) => (
                <ListItem
                  onClick={(e) => handleHashScroll(e, explore.isHash ?? false)}
                  key={explore.title}
                  title={explore.title}
                  href={explore.href}
                >
                  {explore.description}
                </ListItem>
              ))}
            </ul>
          </NavigationMenuContent>
        </NavigationMenuItem>
      </NavigationMenuList>
    </NavigationMenu>
  );
}

const ListItem = React.forwardRef<
  React.ElementRef<"a">,
  React.ComponentPropsWithoutRef<"a"> & { title: string; image?: string }
>(({ className, title, image, children, ...props }, ref) => {
  return (
    <li>
      <NavigationMenuLink asChild>
        <a
          ref={ref}
          className={cn(
            "block select-none space-y-1 rounded-md p-3 leading-none no-underline outline-none !border-none transition-colors hover:bg-accent hover:text-accent-foreground focus:bg-accent focus:text-accent-foreground",
            className,
            itemClassName
          )}
          {...props}
        >
          <div className="text-base font-bold leading-none flex items-center gap-x-2.5">
            {image && (
              <Image
                src={image}
                alt=""
                width={32}
                height={32}
                className="size-6"
              />
            )}
            {title}
          </div>
          <div className="line-clamp-2 text-sm leading-snug text-muted-foreground">
            {children}
          </div>
        </a>
      </NavigationMenuLink>
    </li>
  );
});
ListItem.displayName = "ListItem";
