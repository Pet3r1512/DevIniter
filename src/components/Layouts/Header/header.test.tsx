import { render, screen, waitFor } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import Header from "./_index";

describe("Render Header", () => {
  it("Should render a skeleton", async () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("Should render Header after loading", async () => {
    const { asFragment } = render(<Header />);

    await waitFor(() => screen.getByTestId("header"));

    expect(asFragment()).toMatchSnapshot();
  });
});
