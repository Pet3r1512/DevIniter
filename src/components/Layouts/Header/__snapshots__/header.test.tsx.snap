// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render Header > Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="flex items-center justify-between py-5 px-5 lg:px-0 sticky top-0 z-30 !backdrop-filter !backdrop-blur-xl"
      >
        <div
          class="max-w-[1440px] w-full mx-auto flex items-center justify-between"
        >
          <button
            aria-controls="radix-:r0:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="lg:hidden w-1/3"
            data-state="closed"
            id="sidebar-trigger"
            type="button"
          >
            <svg
              class="lucide lucide-menu"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="4"
                x2="20"
                y1="12"
                y2="12"
              />
              <line
                x1="4"
                x2="20"
                y1="6"
                y2="6"
              />
              <line
                x1="4"
                x2="20"
                y1="18"
                y2="18"
              />
            </svg>
          </button>
          <div
            class="flex items-center gap-x-8"
          >
            <img
              alt="Logo"
              class="lg:h-12 w-auto h-10 max-w-1/3 lg:max-w-none"
              data-nimg="1"
              data-testid="Logo"
              decoding="async"
              height="0"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=3840&q=85"
              srcset="/_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=640&q=85 640w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=750&q=85 750w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=828&q=85 828w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=1080&q=85 1080w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=1200&q=85 1200w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=1920&q=85 1920w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=2048&q=85 2048w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=3840&q=85 3840w"
              style="color: transparent;"
              width="0"
            />
            <nav
              class="hidden lg:flex items-center gap-x-8 mt-1.5"
              data-testid="navbar"
            >
              <a
                class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
                href="/docs/introduction"
              >
                Documents
              </a>
              <a
                class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
                href="https://github.com/Pet3r1512/DevIniter/issues/new?labels=enhancement&template=feature-request---.md"
              >
                Contribution
              </a>
              <a
                class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
                href="#demo"
              >
                Live Demo
                <span
                  class="w-fit !text-[8.5px] bg-[#008000] text-white rounded-lg text-center relative ml-1.5 lg:ml-0 lg:top-[-12px] text-sm px-1.5 py-1"
                >
                  New
                </span>
              </a>
              <a
                class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
                href="#"
              >
                Release Notes
                <span
                  class="w-fit !text-[8.5px] bg-[#e0ac00] dark:bg-[#e09b31] text-white rounded-lg text-center relative ml-1.5 lg:ml-0 lg:top-[-12px] text-sm px-1.5 py-1"
                >
                  Coming
                </span>
              </a>
            </nav>
          </div>
          <div
            class="flex items-center gap-x-5 lg:hidden w-1/3 justify-end"
          >
            <button
              aria-controls="radix-:r3:"
              aria-expanded="false"
              aria-haspopup="dialog"
              class="hidden"
              data-state="closed"
              data-testid="search-trigger"
              type="button"
            >
              <div
                class="flex items-center py-2 px-4 rounded-2xl shadow-2xl lg:dark:bg-gray-dark lg:bg-gray-light lg:dark:text-gray-light text-gray lg:w-48"
              >
                <svg
                  class="lucide lucide-search lg:mr-2"
                  fill="none"
                  height="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <path
                    d="m21 21-4.3-4.3"
                  />
                </svg>
                <span
                  class="hidden lg:block"
                >
                  Search
                </span>
              </div>
            </button>
          </div>
          <div
            class="lg:flex items-center gap-x-5 hidden"
          >
            <button
              class="h-8 lg:h-10 px-5 py-2 w-auto rounded-full lg:text-lg bg-gray-500 dark:bg-gray-200 transition-all dark:text-black text-white duration-150 ease-linear flex items-center justify-center"
              data-testid="theme-togger"
            >
              ðŸŒ™ Dark
            </button>
            <div
              class="border-l-2 border-black-main dark:border-white h-[25px]"
              data-testid="vt"
            />
            <a
              aria-label="Repo Github"
              class="rounded-full size-8 flex items-center justify-center bg-gray-light opacity-65 lg:hover:opacity-100 transition-all duration-150 ease-linear"
              data-testid="github-logo"
              href="https://github.com/Pet3r1512/DevIniter"
              target="_blank"
            >
              <svg
                fill="none"
                height="20"
                viewBox="0 0 256 250"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0_2_15640)"
                >
                  <path
                    d="M128.001 0C57.3173 0 0 57.3067 0 128.001C0 184.555 36.6762 232.536 87.5349 249.461C93.932 250.646 96.2806 246.684 96.2806 243.303C96.2806 240.251 96.1619 230.168 96.1068 219.472C60.4968 227.215 52.9826 204.37 52.9826 204.37C47.16 189.575 38.7704 185.641 38.7704 185.641C27.1569 177.696 39.6458 177.859 39.6458 177.859C52.4993 178.762 59.2674 191.05 59.2674 191.05C70.6838 210.618 89.2116 204.961 96.5159 201.69C97.6647 193.418 100.982 187.771 104.643 184.574C76.2118 181.338 46.3248 170.362 46.3248 121.316C46.3248 107.341 51.3251 95.9224 59.5132 86.9584C58.1842 83.7344 53.8029 70.7156 60.7532 53.0844C60.7532 53.0844 71.5019 49.6442 95.9626 66.205C106.173 63.3689 117.123 61.9466 128.001 61.8978C138.879 61.9466 149.838 63.3689 160.067 66.205C184.498 49.6442 195.232 53.0844 195.232 53.0844C202.199 70.7156 197.816 83.7344 196.487 86.9584C204.694 95.9224 209.66 107.341 209.66 121.316C209.66 170.479 179.716 181.304 151.213 184.473C155.804 188.445 159.895 196.235 159.895 208.177C159.895 225.303 159.747 239.087 159.747 243.303C159.747 246.71 162.051 250.701 168.539 249.444C219.37 232.5 256 184.536 256 128.001C256 57.3067 198.691 0 128.001 0ZM47.9406 182.34C47.6586 182.976 46.6582 183.167 45.7467 182.73C44.8183 182.313 44.2969 181.446 44.5979 180.808C44.8734 180.153 45.876 179.97 46.8023 180.409C47.7328 180.827 48.2627 181.702 47.9406 182.34ZM54.2368 187.958C53.6263 188.524 52.433 188.261 51.6233 187.367C50.786 186.475 50.6292 185.281 51.2481 184.707C51.8776 184.141 53.035 184.406 53.8743 185.298C54.7116 186.201 54.8748 187.386 54.2368 187.958ZM58.5562 195.146C57.772 195.691 56.4896 195.18 55.6968 194.042C54.9126 192.904 54.9126 191.539 55.7138 190.992C56.5087 190.445 57.772 190.937 58.5753 192.066C59.3575 193.224 59.3575 194.589 58.5562 195.146ZM65.8614 203.471C65.1598 204.245 63.6654 204.037 62.5717 202.982C61.4525 201.949 61.1409 200.485 61.8446 199.711C62.5547 198.935 64.0575 199.153 65.1598 200.201C66.2705 201.231 66.6096 202.706 65.8614 203.471ZM75.3025 206.282C74.993 207.284 73.5538 207.74 72.104 207.314C70.6563 206.875 69.7088 205.701 70.0013 204.688C70.3023 203.679 71.7479 203.204 73.2083 203.66C74.6539 204.096 75.6035 205.262 75.3025 206.282ZM86.0469 207.474C86.083 208.529 84.8536 209.405 83.3317 209.424C81.8013 209.458 80.5634 208.603 80.5465 207.565C80.5465 206.499 81.7483 205.632 83.2787 205.606C84.8006 205.577 86.0469 206.424 86.0469 207.474ZM96.6021 207.069C96.7844 208.099 95.7267 209.157 94.2154 209.439C92.7296 209.71 91.3539 209.074 91.1653 208.053C90.9809 206.997 92.0576 205.939 93.5414 205.666C95.0548 205.403 96.4093 206.022 96.6021 207.069Z"
                    fill="#161614"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0_2_15640"
                  >
                    <rect
                      fill="white"
                      height="250"
                      width="256"
                    />
                  </clippath>
                </defs>
              </svg>
            </a>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="flex items-center justify-between py-5 px-5 lg:px-0 sticky top-0 z-30 !backdrop-filter !backdrop-blur-xl"
    >
      <div
        class="max-w-[1440px] w-full mx-auto flex items-center justify-between"
      >
        <button
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-haspopup="dialog"
          class="lg:hidden w-1/3"
          data-state="closed"
          id="sidebar-trigger"
          type="button"
        >
          <svg
            class="lucide lucide-menu"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="4"
              x2="20"
              y1="12"
              y2="12"
            />
            <line
              x1="4"
              x2="20"
              y1="6"
              y2="6"
            />
            <line
              x1="4"
              x2="20"
              y1="18"
              y2="18"
            />
          </svg>
        </button>
        <div
          class="flex items-center gap-x-8"
        >
          <img
            alt="Logo"
            class="lg:h-12 w-auto h-10 max-w-1/3 lg:max-w-none"
            data-nimg="1"
            data-testid="Logo"
            decoding="async"
            height="0"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=3840&q=85"
            srcset="/_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=640&q=85 640w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=750&q=85 750w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=828&q=85 828w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=1080&q=85 1080w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=1200&q=85 1200w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=1920&q=85 1920w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=2048&q=85 2048w, /_next/image?url=%2Fimages%2Flogos%2FFullLogo.png&w=3840&q=85 3840w"
            style="color: transparent;"
            width="0"
          />
          <nav
            class="hidden lg:flex items-center gap-x-8 mt-1.5"
            data-testid="navbar"
          >
            <a
              class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
              href="/docs/introduction"
            >
              Documents
            </a>
            <a
              class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
              href="https://github.com/Pet3r1512/DevIniter/issues/new?labels=enhancement&template=feature-request---.md"
            >
              Contribution
            </a>
            <a
              class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
              href="#demo"
            >
              Live Demo
              <span
                class="w-fit !text-[8.5px] bg-[#008000] text-white rounded-lg text-center relative ml-1.5 lg:ml-0 lg:top-[-12px] text-sm px-1.5 py-1"
              >
                New
              </span>
            </a>
            <a
              class="lg:hover:text-primary dark:text-white transition-all duration-150 ease-linear lg:text-lg relative text-gray"
              href="#"
            >
              Release Notes
              <span
                class="w-fit !text-[8.5px] bg-[#e0ac00] dark:bg-[#e09b31] text-white rounded-lg text-center relative ml-1.5 lg:ml-0 lg:top-[-12px] text-sm px-1.5 py-1"
              >
                Coming
              </span>
            </a>
          </nav>
        </div>
        <div
          class="flex items-center gap-x-5 lg:hidden w-1/3 justify-end"
        >
          <button
            aria-controls="radix-:r3:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="hidden"
            data-state="closed"
            data-testid="search-trigger"
            type="button"
          >
            <div
              class="flex items-center py-2 px-4 rounded-2xl shadow-2xl lg:dark:bg-gray-dark lg:bg-gray-light lg:dark:text-gray-light text-gray lg:w-48"
            >
              <svg
                class="lucide lucide-search lg:mr-2"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <path
                  d="m21 21-4.3-4.3"
                />
              </svg>
              <span
                class="hidden lg:block"
              >
                Search
              </span>
            </div>
          </button>
        </div>
        <div
          class="lg:flex items-center gap-x-5 hidden"
        >
          <button
            class="h-8 lg:h-10 px-5 py-2 w-auto rounded-full lg:text-lg bg-gray-500 dark:bg-gray-200 transition-all dark:text-black text-white duration-150 ease-linear flex items-center justify-center"
            data-testid="theme-togger"
          >
            ðŸŒ™ Dark
          </button>
          <div
            class="border-l-2 border-black-main dark:border-white h-[25px]"
            data-testid="vt"
          />
          <a
            aria-label="Repo Github"
            class="rounded-full size-8 flex items-center justify-center bg-gray-light opacity-65 lg:hover:opacity-100 transition-all duration-150 ease-linear"
            data-testid="github-logo"
            href="https://github.com/Pet3r1512/DevIniter"
            target="_blank"
          >
            <svg
              fill="none"
              height="20"
              viewBox="0 0 256 250"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_2_15640)"
              >
                <path
                  d="M128.001 0C57.3173 0 0 57.3067 0 128.001C0 184.555 36.6762 232.536 87.5349 249.461C93.932 250.646 96.2806 246.684 96.2806 243.303C96.2806 240.251 96.1619 230.168 96.1068 219.472C60.4968 227.215 52.9826 204.37 52.9826 204.37C47.16 189.575 38.7704 185.641 38.7704 185.641C27.1569 177.696 39.6458 177.859 39.6458 177.859C52.4993 178.762 59.2674 191.05 59.2674 191.05C70.6838 210.618 89.2116 204.961 96.5159 201.69C97.6647 193.418 100.982 187.771 104.643 184.574C76.2118 181.338 46.3248 170.362 46.3248 121.316C46.3248 107.341 51.3251 95.9224 59.5132 86.9584C58.1842 83.7344 53.8029 70.7156 60.7532 53.0844C60.7532 53.0844 71.5019 49.6442 95.9626 66.205C106.173 63.3689 117.123 61.9466 128.001 61.8978C138.879 61.9466 149.838 63.3689 160.067 66.205C184.498 49.6442 195.232 53.0844 195.232 53.0844C202.199 70.7156 197.816 83.7344 196.487 86.9584C204.694 95.9224 209.66 107.341 209.66 121.316C209.66 170.479 179.716 181.304 151.213 184.473C155.804 188.445 159.895 196.235 159.895 208.177C159.895 225.303 159.747 239.087 159.747 243.303C159.747 246.71 162.051 250.701 168.539 249.444C219.37 232.5 256 184.536 256 128.001C256 57.3067 198.691 0 128.001 0ZM47.9406 182.34C47.6586 182.976 46.6582 183.167 45.7467 182.73C44.8183 182.313 44.2969 181.446 44.5979 180.808C44.8734 180.153 45.876 179.97 46.8023 180.409C47.7328 180.827 48.2627 181.702 47.9406 182.34ZM54.2368 187.958C53.6263 188.524 52.433 188.261 51.6233 187.367C50.786 186.475 50.6292 185.281 51.2481 184.707C51.8776 184.141 53.035 184.406 53.8743 185.298C54.7116 186.201 54.8748 187.386 54.2368 187.958ZM58.5562 195.146C57.772 195.691 56.4896 195.18 55.6968 194.042C54.9126 192.904 54.9126 191.539 55.7138 190.992C56.5087 190.445 57.772 190.937 58.5753 192.066C59.3575 193.224 59.3575 194.589 58.5562 195.146ZM65.8614 203.471C65.1598 204.245 63.6654 204.037 62.5717 202.982C61.4525 201.949 61.1409 200.485 61.8446 199.711C62.5547 198.935 64.0575 199.153 65.1598 200.201C66.2705 201.231 66.6096 202.706 65.8614 203.471ZM75.3025 206.282C74.993 207.284 73.5538 207.74 72.104 207.314C70.6563 206.875 69.7088 205.701 70.0013 204.688C70.3023 203.679 71.7479 203.204 73.2083 203.66C74.6539 204.096 75.6035 205.262 75.3025 206.282ZM86.0469 207.474C86.083 208.529 84.8536 209.405 83.3317 209.424C81.8013 209.458 80.5634 208.603 80.5465 207.565C80.5465 206.499 81.7483 205.632 83.2787 205.606C84.8006 205.577 86.0469 206.424 86.0469 207.474ZM96.6021 207.069C96.7844 208.099 95.7267 209.157 94.2154 209.439C92.7296 209.71 91.3539 209.074 91.1653 208.053C90.9809 206.997 92.0576 205.939 93.5414 205.666C95.0548 205.403 96.4093 206.022 96.6021 207.069Z"
                  fill="#161614"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_2_15640"
                >
                  <rect
                    fill="white"
                    height="250"
                    width="256"
                  />
                </clippath>
              </defs>
            </svg>
          </a>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
